engine/benches/bench.rs

```diff
@@ -44,42 +65,66 @@ impl<'a, T: 'static + Copy + Debug + Into<LhsValue<'static>>> FieldBench<'a, T>
 
             c.bench(
                 "parsing",
-                Benchmark::new(name, move |b: &mut Bencher| {
-                    let mut scheme = Scheme::default();
-                    scheme.add_field(field.to_owned(), ty).unwrap();
+                Benchmark::new(name, {
+                    let funcs = functions.to_vec();
```

Comment:
Instead of creating an intermediate vector, maybe move out entire Scheme initialisation from inside the closure? (not sure whether that will work well with lifetimes, but I don't see any reason why not)

Reply:
Well the `Scheme` would have to outlive the `run` function. I guess we could have a global `lazy_static` scheme for all benchs but that would prevent us from having a specific scheme per bench. Anyway its a bit outside of the scope of this PR.

Reply:
> Well the Scheme would have to outlive the run function.

Hmm, no, why?

Reply:
It's the same as with current vector - when you use `move |...|`, it moves the variable into the closure.

Reply:
Ok got it :+1:
I updated the PR